### ISSUE TITLE
menu: focus lost when opening with keyboard and first item is disabled

### ISSUE THREAD

**Issue by SemLR** [angular/material#12054]
## Bug Report
When first item from an md-menu-content is disabled, the item container is not getting focus. Forbidding keyboard-only usage.

## Demo and steps to reproduce the issue

[Codepen demo](https://codepen.io/Laurenzo/pen/bGwwJMz)

#### Detailed Reproduction Steps
1. Open Above link
2. Click near first phone icon
3. Press tab key to give focus to md-menu 
4. Press Enter to open menu
5. Try to reach one of enabled button

### Explain the expected behavior
When opening via keyboard an md-menu, Focus should be get by first Button which is not disabled, even if md-content is filled with custom menu-item.

### Explain the current behavior
Focus seems do be kept by menu button if first md-menu custom item is disabled.

### List the affected versions of AngularJS, Material, OS, and browsers
- AngularJS: 1.7.9 to current
- AngularJS Material: from 1.1.20 to current 
- OS: GnuLinux Debian based distrib
- Browsers: Chromium

**Comment by Splaktar** [comment]
This seems to work properly when using `md-button` inside of your `md-menu-item`s. [Working Demo](https://codepen.io/Splaktar/pen/yLagqpz?editors=1010)

I'm not sure what library this is from:
```html
          <a-menu-action label="item" is-disabled="$index == 0">
            </a-menu-action>
```
But it doesn't appear to be compatible with `md-menu` without some custom styling (width is too short) and event handling (md-menu's built in focus handling requires `md-button` elements).

**Comment by SemLR** [comment]
@Splaktar thank you for your attention to this problem.
The component that is used is not part of a specific library and is declared alongside in the codepen. This Codepen is the simplest example showing the problem we are facing, when using components other than md-menu-item. 
I confirm that the component works well with menu-item but not with other components. 

The documentation clearly states: "Typically this will contain md-menu-items, but you can do custom content as well."

The  provided component relies internally on an md-button, whose focus is correctly recovered if it is first and non-disabled, but not if after a disabled md-button in the list. 
Does it make sense, and am i clear enough to point the problem ?

**Comment by Splaktar** [comment]
Thank you for the heads up on that custom component, I didn't see it there at the bottom of the JS in CodePen.

I've updated my [demo](https://codepen.io/Splaktar/pen/rNMyWaN?editors=1010) to get the styling fixed and to use an approach that we can support (`ng-disabled` on first child element of the `md-menu-item`).

Currently, if we don't find a non-disabled `.md-button` in the first `md-menu-item`, we just check if it has a child element and that the child element's `disabled` property is falsy. However, this only works for HTML elements that have a disabled property (like `button`) and doesn't work for custom elements (AngularJS Components).

In PR #12058, I've added support for checking the `disabled` attribute as well. This allows you to set `ng-disabled` or `disabled` on your `<a-menu-action>` element so that it will be skipped in the search for the first focusable `md-menu-item` after the menu is opened.

**Comment by SemLR** [comment]
Thank you for your attention and your responsiveness.
